data-preprocessing/JSON_Section_Extractor/json_section_extractor.py

```diff
@@ -0,0 +1,130 @@
+                    pages.append(page)
+                    sections.append(str(" ".join(section)))
+            except:
+                continue
+
+
+def create_directory(directory):
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    else:
+        pass
+
+    # Open the CSV file in write mode
+def write_sections_to_txt(file, list1, list2, directory):
+    # to obtain the filename without the end extension '.json', the last five(5) characters from the filename must be stripped off...
+    # for eg: Algebra.json can be changed to Algebra by using Algebra.json[:-5], with the -5 taking care of the last five characters.
+    filename = f"{file[:-5]}.txt"
+    file_path = os.path.join(directory, filename)
+    with open(file_path, 'w') as file:
+        for item1, item2 in zip(list1, list2):
+            file.write(f"{item1}\t{item2}\n")
+
+
+class ScriptUsage:
+    def __init__(self):
+        self.parser = argparse.ArgumentParser()
+
+    def version(self):
+        print("Script version: 1.0")
+
+    def instruction(self):
+        print("Script instructions:")
+        print("1. Provide the absolute path to the json files in the booksPath.json file.")
+        print("2. Make sure the files in the path provided end with an json extension.")
+        print("3. Provide the name of the directory that should be created in the root directory, in which results will be stored.")
+
+    def help(self):
+        self.parser.print_help()
+    
+    def main(self):
+        parsed_books_path = read_PATH_json()['path_to_parsed_books']
+        results_path = read_PATH_json()['directory']
+        all_json_files = put_json_filenames_in_list(parsed_books_path)
+
+        for file in all_json_files:
+            pages = []
+            sections = []
+            json_dict = load_json_file(file, parsed_books_path)
+
+
+            json_to_section(json_dict,pages, sections, nested_list=True)
+
+            # second json structure
+            if len(sections) == 0: # checks if all sections are empty strings. If yes, there is a tendency that the current json contain direct objects and not nested lists.
+                json_to_section(json_dict, pages, sections) # return all sections using the second json structure as the standard.
+
+            # write results to csv file
+            create_directory(results_path)
+            write_sections_to_txt(file, pages, sections, results_path)
+
+
+
+    def parse_args(self):
+        self.parser.add_argument('-v', '--version', action='store_true', help='Display script version')
+        self.parser.add_argument('-i', '--instruction', action='store_true', help='Display script instructions')
+        args = self.parser.parse_args()
+
+        if args.version:
+            self.version()
+        elif args.instruction:
+            self.instruction()
+        else:
+            self.main()
+
+
+if __name__ == '__main__':
+    usage = ScriptUsage()
+    usage.parse_args()
+
+    
```

Comment:
add a newline to get rid of the red error. if you hover over it you'll see the error

Reply:
Please I do not see a error

Reply:
Please I do not see a error

Reply:
Can you please help mm figure it out?

Reply:
![image](https://github.com/nsmq-ai/nsmqai/assets/10053339/62dce1ba-2152-4b00-83c6-6737cb8e320b)

the red circle is the error I am referring to. Once you add a newline to the end of the file it will go away